src/agent/thread_helper.cpp

```diff
@@ -326,15 +326,25 @@ void ThreadHelper::sJoinerCallback(otError aError, void *aThreadHelper)
 
 void ThreadHelper::JoinerCallback(otError aError)
 {
-    if (aError != OT_ERROR_NONE)
+    if (aError == OT_ERROR_NONE)
     {
-        otIp6SetEnabled(mInstance, false);
-        mJoinerHandler(aError);
-        mJoinerHandler = nullptr;
+        aError = otThreadSetEnabled(mInstance, true);
     }
     else
     {
-        otThreadSetEnabled(mInstance, true);
+        otError error = otIp6SetEnabled(mInstance, false);
+
+        if (error != OT_ERROR_NONE)
+        {
+            otbrLog(OTBR_LOG_WARNING, "Failed to stop Thread network: %s", otThreadErrorToString(error));
+        }
+    }
+
+    if (aError != OT_ERROR_NONE)
```

Comment:
merge into `else {...}` above? and we should log "Failed to join Thread network..." before "Failed to stop Thread network...".

Reply:
This also logs error in line 331.

Reply:
ok, better don't change pass-in `aError`.

Currently, the logging order is not correct if both log messages are printed. 

How about we log `aError` at the beginning of this function and log result of `otIp6SetEnabled` separately?